web/pdf_viewer.css

```diff
@@ -46,6 +46,46 @@
   border: none;
 }
 
+.pdfViewer.scrollHorizontal, .pdfViewer.scrollWrapped, .spread {
+  text-align: center;
+}
+
+.pdfViewer.scrollHorizontal, .pdfViewer.scrollWrapped {
+  margin: 4px;
```

Comment:
Is this necessary? Why can't this be removed, and then lines 78-79 be combined with line 74?

Reply:
Because then there isn't enough space to the left of the first page in horizontal scrolling, if you did literally that.

The real answer to your question is that the ‘true’ margin of `.pdfViewer` is meant to be 4px, even in vertical scrolling, and the ‘true’ ‘effective margin’ of pages is meant to be `6px 6.5px 5px 6.5px` in all cases (‘effective margin’ meaning margin + border, which is what the user sees as the spacing between pages). Really, what I think one should be asking is why the `.scrollHorizontal`/`.scrollVertical` constraints are here at all, and why this isn't just the margin in all scroll modes (in which case, pages would get the margins in line 80 in all modes as well). The answer to that question is that when I tried that, in my testing, there was more space between pages in vertical scrolling mode than I expected. When the page elements have `display: block`, thanks to margin collapsing, despite negative margins on both pages, the space between the two pages is determined by the border of the top page plus the border of the bottom page minus only one of the two negative margins. The original CSS seems to get around this by having only one of the top/bottom margins be negative, so I continued that pattern. In the other two scrolling modes, pages have `display: inline-block`, which doesn't trigger margin collapsing even when the pages wrap onto the next line.

So that's why there are basically two sets of margin values used for `.pdfViewer` and `.page`: one for vertical scrolling with `display: block`, and one for the other scrolling modes with `display: inline-block`, the latter of which is what you're commenting on here.